git/setup/action.yml

```diff
@@ -0,0 +1,42 @@
+---
+name: 'Configure git CLI'
+description: |
+  Setup the git usename, email, and authentication with git CLI.
+  After this action you can use https URL repos in git commands.
+  The GITHUB_TOKEN (not needed here) will be used to authenticate with the git CLI,
+  after this action.
+inputs:
+  username:
+    description: 'Git username.'
+    required: false
+    default: 'obltmachine'
+  email:
+    description: 'Git email.'
+    required: false
+    default: 'obltmachine@users.noreply.github.com'
+  trace:
+    description: 'Enable git trace.'
+    required: false
+    default: 'false'
+  token:
+    description: 'GitHub token.'
+    default: ${{ github.token }}
+    required: false
+runs:
+  using: "composite"
+  steps:
+    - name: Setup git
+      shell: bash
+      run: |
+        git config --global user.name "${GIT_USER}"
+        git config --global user.email "${GIT_EMAIL}"
+        gh auth setup-git
+
+        echo "GIT_USER=${GIT_USER}" >> $GITHUB_ENV
+        echo "GIT_EMAIL=${GIT_EMAIL}" >> $GITHUB_ENV
```

Comment:
Can this be converted to outputs or this only make sense as environment variables?

Reply:
For simplicity I think env variables, then consumers don't need to care about creating them after using this opinionated action